### PR TITLE
cmdstan: unbreak on aarch64-linux

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

The package also had 1 Hydra failures on aarch64-darwin (where no binaries are provided), so #199919.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
Result of `nixpkgs-review pr 203381` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>cmdstan</li>
  </ul>
</details>

**Comment by wegank** [comment]
@wamserma Sorry for having x86_64 emulation enabled on my machine! Could you please review again?

**Comment by wamserma** [comment]
Result of `nixpkgs-review pr 203381` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cmdstan</li>
  </ul>
</details>

File: pkgs/development/compilers/cmdstan/default.nix [hunk 1]
```diff
@@ -44,7 +58,6 @@ stdenv.mkDerivation rec {
   preFixup = "rm -rf $(pwd)";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/cmdstan/default.nix:58]: That looks dangerous and if we are in the wrong directory could just delete everything

File: pkgs/development/compilers/cmdstan/default.nix [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+  srcs = rec {
+    aarch64-linux = fetchurl {
+      url = "https://github.com/stan-dev/cmdstan/releases/download/v${version}/cmdstan-${version}-linux-arm64.tar.gz";
+      sha256 = "sha256-oj/7JHT4LZcRAHiA2KbM6pZbOe6C98Ff//cNsG9DIm8=";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://github.com/stan-dev/cmdstan/releases/download/v${version}/cmdstan-${version}.tar.gz";
+      sha256 = "sha256-BMqRRWIC/Z7It2qkESJd9L3ycyxvA6NHiWbAvzVMzIQ=";
+    };
+    x86_64-linux = x86_64-darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/cmdstan/default.nix:16]: Shouldn't we then differentiate between arch instead?

Comment (SuperSandro2000) [reply pkgs/development/compilers/cmdstan/default.nix:16]: Okay thats strange but then we leave it as is

File: pkgs/development/compilers/cmdstan/default.nix [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
   };
+  src = srcs.${stdenv.hostPlatform.system};
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/cmdstan/default.nix:18]: Missing or throw unsupported system otherwise this is a syntax error on other paltforms.

Comment (wegank) [reply pkgs/development/compilers/cmdstan/default.nix]: Done.

**Comment by SuperSandro2000** [comment]
For the next time: If you don't rebase on master before force pushing it is easier for me to see what exactly you changed and then merge it immediately.

**Pull request merged**